### DISCUSSION TITLE
x265 mods

### DISCUSSION THREAD

**Discussion by Dendraspis** [staxrip/staxrip#479] [General]
Meanwhile the benefits of each mod were even removed from the wiki.

So is there meanwhile any difference between those versions from Patman86, DJATOM and msg7086 apart from the VPY direct input buffer issue, that surely will be adopted by the others soon?

I have to admit that I lost track and it seems like they would be almost the same, but developed on 3 different repositories.  🙄

**Comment by ghost** [comment]
@Dendraspis 
Yes there are differences: since a couple of days, Patman86's build will not be usable in staxrip because it doesn't have the --reader-options library switch.
I was sad about that :-(

Currently, the build made by **DJATOM is the only that works in staxrip.**
I believe the commits will follow soon at msg7086's.
And eventually, if Patman86 has time or wishes to do it, he will implement it, but I don't think it's gonna be soon

**Reply by Dendraspis** [reply]
In other words:  One is still ahead and you have to wait till the others follow on or switch to the one which is most ahead.
So I wonder if someday they are going to decide to merge their versions into one repository to save time maintaining them.

Yeah, I also liked Patman86's modded versions. But since the direct input I like DJATOM's mod, too, especially with those expanded progress information.  😃

**Reply by DJATOM** [reply]
Probably it's not so difficult to have workarounds on known mods/versions. For example, check mod name and version, then use certain codepath for it. For example, Yuuki-Asuna mod can provide external frameserver library, but it uses different scheme for it (extract it from filename, which is not so convenient, I think). My approach allows implementing any number of reader-specific options without further edits of x265 core code. Maybe I'll implement an option to pass json files into `reader-options`, as there's already json reader present in codebase. For now I want to ensure there is no bugs with vpy input. Btw, I'm tinkering with x264 code to provide aMod of latest x264 version. Got working lavf, ffms2 readers, external library loading for avs and mp4 (lsmash) muxers. It took mostly all night to make fully working ffmpeg libs with ffms2 (I had wrong zlib [from msys] in my environment and it polluted the whole ffmpeg/ffms2 stuff, while custom build from sources worked fine).

**Reply by Patman86** [reply]
> And eventually, if Patman86 has time or wishes to do it, he will implement it, but I don't think it's gonna be soon

I have no intention of compiling new builds for x264 and x265.  DJATOM's works perfectly!

**Reply by ghost** [reply]
We'll always remember how we were so fond of your x265 builds @Patman86 ❤️ 
That will now remove some load on your shoulders and allow you spend more time with your family :=)

**Reply by Patman86** [reply]
Thank you for the kind words. It's not that I'm not doing anything. For me it remains a hobby and since I have the time I will still do builds and publish them. I think it will be a mix of the msg7086 and DJATOM repositories as these have a lot of good features. I can't give an exact date yet.

**Reply by DJATOM** [reply]
Mentioned x264 mod got a release. Test it, report bugs, etc.
